api-server/api-server-common/src/storage/impls/in_memory/transactional/read.rs

```diff
@@ -30,6 +30,13 @@ impl<'t> ApiServerStorageRead for ApiServerInMemoryStorageTransactionalRo<'t> {
         self.transaction.is_initialized()
     }
 
+    async fn get_address_balance(
+        &self,
+        address: &str,
+    ) -> Result<Option<Amount>, ApiServerStorageError> {
```

Comment:
Please understand that the balance isn't just in coins, but also in tokens. So this should return a map of coins and tokens, similar to AmountsMap in TransactionVerifier. You don't have to do this right now, but it can be added to the list.

Reply:
Yep, there's already a placeholder in the API endpoint for tokens, just not implemented yet. To lesson the complexity though, I was going to add token balances to a new function.